java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java

```diff
@@ -34,20 +35,67 @@
 import org.sonar.plugins.java.api.tree.MethodInvocationTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
+import javax.annotation.CheckForNull;
+
+import java.util.Arrays;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 
+import static org.sonar.java.checks.DeprecatedHashAlgorithmCheck.InsecureAlgorithm.MD2;
+import static org.sonar.java.checks.DeprecatedHashAlgorithmCheck.InsecureAlgorithm.MD5;
+import static org.sonar.java.checks.DeprecatedHashAlgorithmCheck.InsecureAlgorithm.SHA1;
+
 @Rule(key = "S2070")
 public class DeprecatedHashAlgorithmCheck extends AbstractMethodDetection {
 
   private static final String JAVA_LANG_STRING = "java.lang.String";
-  private static final String MD5 = "MD5";
-  private static final String SHA1 = "SHA1";
 
-  private static final Map<String, String> ALGORITHM_BY_METHOD_NAME = ImmutableMap.<String, String>builder()
+  enum InsecureAlgorithm {
+    MD2, MD4, MD5, MD6, RIPEMD160,
+    SHA1 {
+      @Override
+      public String toString() {
+        return "SHA-1";
+      }
+    },
+    DSA {
+      @Override
+      boolean match(String algorithm) {
+        // exact match required for DSA, so it doesn't match ECDSA
+        return "DSA".equals(algorithm);
+      }
+    };
+
+    boolean match(String algorithm) {
+      String normalizedName = algorithm.replaceAll("-", "").toLowerCase(Locale.ROOT);
```

Comment:
Use `Locale.ENGLISH` (the one which should be used everywhere, don't know why Locale.ROOT is used in some other parts).